subsys/bluetooth/controller/ll_sw/ull_master.c

```diff
@@ -51,32 +52,67 @@
 
 static void ticker_op_stop_scan_cb(u32_t status, void *params);
 static void ticker_op_cb(u32_t status, void *params);
-static inline void access_addr_get(u8_t access_addr[]);
 static inline void conn_release(struct ll_scan_set *scan);
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+u8_t ll_create_connection(u16_t scan_interval, u16_t scan_window,
+			  u8_t filter_policy, u8_t peer_addr_type,
+			  u8_t *peer_addr, u8_t own_addr_type,
+			  u16_t interval, u16_t latency, u16_t timeout,
+			  u8_t phy)
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
 u8_t ll_create_connection(u16_t scan_interval, u16_t scan_window,
 			  u8_t filter_policy, u8_t peer_addr_type,
 			  u8_t *peer_addr, u8_t own_addr_type,
 			  u16_t interval, u16_t latency, u16_t timeout)
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
 {
 	struct lll_conn *conn_lll;
 	struct ll_scan_set *scan;
 	u32_t conn_interval_us;
 	struct lll_scan *lll;
 	struct ll_conn *conn;
 	memq_link_t *link;
-	u8_t access_addr[4];
 	u8_t hop;
 	int err;
 
-	scan = ull_scan_is_disabled_get(0);
+	scan = ull_scan_is_disabled_get(SCAN_HANDLE_1M);
 	if (!scan) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	struct ll_scan_set *scan_coded;
+	struct lll_scan *lll_coded;
+
+	scan_coded = ull_scan_is_disabled_get(SCAN_HANDLE_PHY_CODED);
+	if (!scan_coded) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
 	lll = &scan->lll;
+	lll_coded = &scan_coded->lll;
+
+	if (phy & BT_HCI_LE_EXT_SCAN_PHY_CODED) {
+		if (!lll_coded->conn) {
+			lll_coded->conn = lll->conn;
+		}
+		scan = scan_coded;
+		lll = lll_coded;
+	} else {
+		if (!lll->conn) {
+			lll->conn = lll_coded->conn;
+		}
+	}
+
+	lll->phy = phy;
+
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
+	lll = &scan->lll;
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+
 	if (lll->conn) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
+		goto conn_is_valid;
```

Comment:
This seems to change semantics for the case that `lll->conn!=0` without advertisement extensions enabled (calling `ull_scan_enable` instead of returning `BT_HCI_ERR_CMD_DISALLOWED`). Is this really intended?

Reply:
The change is intentional. The `BT_HCI_ERR_CMD_DISALLOWED` was not exercised due to earlier check on scanning/initiating being in disabled state, hence does not impact the sematics in practice. The new intention here is to use single connection context in two scan set (1M, Coded PHY) and it be only initialized once on the first call to ll_create_connection of 1M and/or Coded PHY from the hci.c, the later call reuses/by-passes initialization.